include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp

```diff
@@ -307,6 +315,15 @@ SpatialOperator<dim, Number>::prescribe_initial_conditions(BlockVectorType & dst
   dst.block(block_index_velocity) = velocity_double;
 }
 
+template<int dim, typename Number>
+void
+SpatialOperator<dim, Number>::set_integrated_rhs(VectorType const & integrated_rhs_in)
+{
+  integrated_rhs_set = true;
+  integrated_rhs.reset(integrated_rhs_in);
+  integrated_rhs->update_ghost_values();
```

Comment:
here I do not understand why we need an update of ghost values?

Reply:
Good catch. We don't need it since we simply add vectors and do not use this in a matrix free loop.

Reply:
[a029183](https://github.com/exadg/exadg/pull/635/commits/a0291830aa127f73a2535204b3b90af41e508557)